projects/sandbox/infer/infer/data/batcher.py

```diff
@@ -1,11 +1,10 @@
-from typing import Iterator
-
 import numpy as np
+from infer.data import Injector
 from ratelimiter import RateLimiter
 
 
 def batch_chunks(
-    it: Iterator,
+    it: Injector,
```

Comment:
Pedantic, but this doesn't need to be an `Injector` class: all it needs to be is an `Iterator` that returns two arrays (so `Iterator[np.ndarray, np.ndarray]` would probably be the most precise. How those arrays get generated is not this function's concern (though given the context it's maybe more helpful to specify precisely which iterator we typically expect)

Reply:
Actually going through and reading the `Injector` code, this will actually never be an `Injector`: we `map` the `Injector` onto an `Iterable` to create an `Iterator` that returns the `Injector`'s action on each element of that `Iterable`. If someone were e.g. using `mypy`, it would raise a flag that `Injector` does not have a `__next__` method.

Reply:
Ah, okay, I didn't understand how the mapping worked.